src/pantry.js

```diff
@@ -1,7 +1,75 @@
 class Pantry {
-  constructor(userIngredients) {
-    this.contents = userIngredients;
+  constructor(userIngredients, ingredientsData) {
+    this.contents = userIngredients,
+    this.ingredientsData = ingredientsData
   }
+
+  compareRecipeToPantryIngredients(chosenRecipe) {
+
+    let ingredientDetails = chosenRecipe.ingredients.map(ingredient => {
+      return {
+        name: ingredient.name,
+        id: ingredient.id,
+        qty: ingredient.quantity.amount,
+        unit: ingredient.quantity.unit,
+        cost: this.ingredientsData.find(ingredientInfo => {
+          return ingredientInfo.id === ingredient.id
+        }).estimatedCostInCents
+      };
+    });
+
+    ingredientDetails.forEach(ingredient => {
+      let pantryMatch = this.contents.find(item => {
+        if (ingredient.id === item.ingredient) {
+          ingredient.qty -= item.amount
+        };
+      });
+    });
+
+    let ingredientsNeeded = ingredientDetails.reduce((need, ingr) => {
+      if (ingr.qty > 0) {
+        need.push(ingr)
+      }
+      return need
+    }, []);
+    return ingredientsNeeded;
+  }
```

Comment:
This looks good! Eventually, we can think about refactoring this into multiple helper functions for the sake of readability.